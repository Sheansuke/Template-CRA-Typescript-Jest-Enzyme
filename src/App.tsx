import React from "react";

interface User {
  name?: String;
}

function App(props: User) {
  return (
    <>
      Test Template
      <div>{props.name}</div>
    </>
  );
}

export default App;
