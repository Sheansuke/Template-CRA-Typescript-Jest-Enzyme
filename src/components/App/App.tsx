import React, { useState } from "react";

interface Props {
  name?: string;
}

const App: React.FC<Props> = () => {
  const [count, setCount] = useState(0);
  return (
    <>
      <div>{count}</div>
      <button type="button" onClick={() => setCount(count + 1)}>
        Increment
      </button>
      <button type="button" onClick={() => setCount(count - 1)}>
        Decrement
      </button>
    </>
  );
};

export default React.memo(App);
