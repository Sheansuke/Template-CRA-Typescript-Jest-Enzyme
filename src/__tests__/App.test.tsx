import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import App from "../components/App/App";

describe("<App />", () => {
  test("counter", () => {
    const component = mount(<App />);
    const button = component.find("button").first();
    button.simulate("click");
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find("div").text()).toEqual("1");
  });
});
