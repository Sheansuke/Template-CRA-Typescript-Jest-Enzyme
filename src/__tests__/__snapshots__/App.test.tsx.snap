// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> counter 1`] = `
<Memo(App)>
  <div>
    1
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    Increment
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    Decrement
  </button>
</Memo(App)>
`;
